etp-core/etp-backend/src/main/clj/solita/etp/api/public_csv.clj

```diff
@@ -0,0 +1,25 @@
+(ns solita.etp.api.public-csv
+  (:require    [ring.util.response :as r]
+               [solita.etp.security :as security]
+               [solita.etp.service.concurrent :as concurrent]
+               [solita.etp.service.csv-to-s3 :as csv-to-s3]
+               [solita.etp.service.kayttaja :as kayttaja-service]))
+
+
+(def internal-routes
+  [["/public-csv"
+    ["/update"
+     {:post {:summary    "Päivitä julkiset CSV-tiedostot S3:ssa."
+             :middleware [[security/wrap-db-application-name
+                           (kayttaja-service/system-kayttaja :aineisto)]
+                          [security/wrap-whoami-for-internal-aineisto-api]]
```

Comment:
Nämä middlewaret voisivat varmaan olla jotain muuta.

Vaikuttavat ainakin niihin, että saako tehdä tietokantakyselyt, ja jos jotain muutettaisiin, niin audit tauluihin tulisi merkintä käyttäjästä.

Koska csv:tä pystyy hakemaan public puolelta, niin voisi varmaan hyppiä siitä routesta taaksepäin ja päätellä, mitä kannattaisi olla, tai tarvitaanko näitä edes🤔

Reply:
Jos nyt tajusin oikein tämän niin routessa takaisinpäin hyppimällä näyttäisi löytyvän publicin alta seuraava middleware, tämä varmaan riittäisi julkisen csvn latauksen tapauksessa?
`  ["/public" {:middleware [[security/wrap-db-application-name]]}`

Reply:
Joo luulisin niin. Lisäksi tuli vähän mietittyä, että se saattaisi vaikutta jopa siihen mitä csv:hen tulee. En nyt ole ihan varma vaikuttaisiko, mutta kai se kulkeutuu tuonne `solita.etp.service.energiatodistus-search/whoami-sql` asti 🤔 

Mutta nyt ainakin pitäisi olla sama. Jos keksii testin, jolla testaa vähän tuota sisältö, niin voisi olla hyvä.